packages/pluggableWidgets/datagrid-web/src/utils/useCellRenderer.tsx

```diff
@@ -0,0 +1,45 @@
+import { createElement, useCallback } from "react";
+import { ListActionValue } from "mendix";
+import { CellRenderer } from "../components/Table";
+import { ColumnsType } from "../../typings/DatagridProps";
+import classNames from "classnames";
+import { executeAction } from "@mendix/piw-utils-internal";
+
+interface CellRendererHookProps {
+    columns: ColumnsType[];
+    onClick?: ListActionValue;
+}
+
+export function useCellRenderer(props: CellRendererHookProps): CellRenderer {
+    const renderer: CellRenderer = (renderWrapper, value, columnIndex) => {
+        const column = props.columns[columnIndex];
+        const title = column.tooltip && column.tooltip.get(value)?.value;
+        let content;
+
+        if (column.showContentAs === "attribute") {
```

Comment:
do we have types for `showContentAs` property? I found that you passing `attribute` and `dynamicText`, but never found the type for `else` case. Do we really need that?

Reply:
We have 3 options in DG2

Reply:
@raffiar, DG2 has option to customize cell content, so the `else` block is about rendering this content